engine/playingsound.cpp

```diff
@@ -0,0 +1,50 @@
+#include "playingsound.h"
+
+namespace pong
+{
+
+PlayingSound::PlayingSound(const Sound& sound) :
+    mSound(&sound),
```

Comment:
can mSound be just saved as a reference?

Reply:
I figured pointers are preferable to references for member variables because they can be initalized to null? Rather than some garbage default value for a reference?

Reply:
ah, yes. And famously std::optional doesn't support references I don't think.

Fine to use pointers when you need a null :) 

Although is it ever valid for a playing sound to have a null sound? that seems like it would be malformed

Reply:
I guess not, I was more worried about there being some garbage default value in the reference in case it did get used before it was initalized.